### PR TITLE
Certificate export - unit tests (EXPOSUREAPP-13330)

### PR DESCRIPTION
**Pull request by mtwalli**
[Ticket](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13330)

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        instance().pdfString.getOrAwaitValue() shouldBe HTML_TEMPLATE.replace(
+            oldValue = "++certificates++",
+            newValue = """
+                <li>Template</li>
+                <li>Template</li>
+                <li>Template</li>
+            """.trimIndent()
+        )
+    }
+
```
Comment (schauersbergern) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewViewModelTest.kt:122]: You could also write a test for sharePDF()

**Review by SamuraiKek — APPROVED** [review]
LGTM

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/helper/CertificateDataInjectorTest.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            every { sampleCollectedAtFormatted } returns "2020-10-20"
+            every { testResult } returns "Negative"
+            every { testCenter } returns "TestIQ"
+            every { testType } returns "Rapid"
+        }
+
+        template(fileName = "template/de_tc_v4.1.svg").inject(tc).contains("\$") shouldBe false
+        template(fileName = "template/tc_v4.1.svg").inject(tc).contains("\$") shouldBe false
+    }
+
```
Comment (schauersbergern) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/helper/CertificateDataInjectorTest.kt:86]: You could also add a test for ```replaceFieldsOf(cert: Cwa...)```

Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/helper/CertificateDataInjectorTest.kt:86]: it is covered by respective Certificate type

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewViewModelTest.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewViewModelTest.kt:122]: will try in the other PR , not so easy to unit test Android APIs in unit tests 🤷‍♂️

**Review by schauersbergern — APPROVED** [review]

**Pull request merged**